dodiscover/constraint/utils.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict, List, Set
+
+import pandas as pd
+
+
+def dummy_sample(G):
+    """Sample an empty dataframe with columns as the nodes.
+
+    Used for oracle testing.
+    """
+    df_values = dict()
+    for node in G.nodes:
+        df_values[node] = []
+
+    df = pd.DataFrame.from_dict(df_values)
```

Comment:
- Could be reduced to ```return pd.DataFrame({column: [] for column in G.nodes})```.
- Note that ```node``` could potentially have a data type that is incompatible with the requirements for pandas column names.

Reply:
Good point, I'll maybe add an error message if not number or string. If we run into errors later by users, I'm sure we can revisit the point.